### PR TITLE
Fix/4224  search fixes 8 17 19 20

### PR DESCRIPTION
**Pull request by grignards**
See #4224 (8th, 17th, 19th and 20th tasks).

## Checkpoints

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)
- [x] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified
- [x] Original authors of the features included in a multi-feature branch (maintenance fixes -> develop, security fixes -> develop, …) should be part of the reviewers, especially if you encountered merge conflicts.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done

### CODE REVIEW

File: frontend/src/container/AdvancedSearch.jsx [hunk 1]
```diff
@@ -297,16 +281,29 @@ export class AdvancedSearch extends React.Component {
...
     const { props } = this
-    const searchObject = parseSearchUrl(qs.parse(props.location.search))
-    if (Object.values(ADVANCED_SEARCH_TYPE).includes(searchObject.searchType)) {
-      this.setState({ searchType: searchObject.searchType })
-      const currentSearch = this.getCurrentSearchObject()
-      this.getSearchResult(searchObject, currentSearch.resultList.length)
+    if (Object.values(ADVANCED_SEARCH_TYPE).includes(searchType)) {
+      this.setState({ searchType: searchType })
     } else {
+      // Default to content search (??)
```
Comment (raphj) [inline frontend/src/container/AdvancedSearch.jsx:300]: Yes.
```suggestion
      // Default to content search
```

**Review by raphj — APPROVED** [review]

File: frontend/src/container/AdvancedSearch.jsx [hunk 2]
```diff
@@ -147,16 +125,22 @@ export class AdvancedSearch extends React.Component {
     const { props } = this
     // INFO - G.B. - 2021-02-12 - check if the user comes through an url that is not placed at first page
     const hasFirstPage = !(currentSearchLength < searchObject.numberResultsByPage * (searchObject.currentPage - 1))
+    const onlyGetFacet = Object.keys(appliedFilters).length === 0 && !searchObject.searchString
+
+    let pageNumber = FIRST_PAGE
+    let pageSize = searchObject.numberResultsByPage
+    if (onlyGetFacet) {
+      pageSize = 0
```
Comment (gbonaspetti) [inline frontend/src/container/AdvancedSearch.jsx:133]: It will be nice to have a comment to explain this 0

Comment (grignards) [reply frontend/src/container/AdvancedSearch.jsx:133]: Done.

**Comment by gbonaspetti** [comment]
After some tests:
- Enter still doesn't trigger a new search
- The simple search has some css problems (I don't know if it is linked to this pr)
Screenshot from 2021-03-09 10-06-32
- I have some errors in the console
Screenshot from 2021-03-09 10-31-01
- There is a weird behavior when I applied the search fields at a "empty" search
Peek 2021-03-09 10-25
- The phrase _No results for the search terms: ""_ should not appears if there are hits
- The date filters does not work when search is "empty"
Peek 2021-03-09 10-26

**Review by gbonaspetti — APPROVED** [review]

**Review by raphj — APPROVED** [review]

**Comment by gbonaspetti** [comment]
After re-test, the following issues are still happening:
- The simple search still have the css problems
- The phrase No results for the search terms: "" should not appears if there are hits
- There is a weird behavior when I applied the search fields at a "empty" search

Peek 2021-03-10 11-12

**Review by gbonaspetti — APPROVED** [review]

**Pull request merged**